src/server/cmd/pachctl/shell/completions.go

```diff
@@ -117,15 +167,14 @@ func FilesystemCompletion(_, text string, maxCompletions int64) []prompt.Suggest
 			Text: filepath.Join(dir, fi.Name()),
 		})
 	}
-	return result
+	return result, func(_, text string) bool {
+		return filepath.Dir(text) == dir
+	}
 }
 
 // PipelineCompletion completes pipeline parameters of the form <pipeline>
-func PipelineCompletion(_, text string, maxCompletions int64) []prompt.Suggest {
-	c, err := client.NewOnUserMachine("user-completion")
-	if err != nil {
-		log.Fatal(err)
-	}
+func PipelineCompletion(_, _ string, maxCompletions int64) ([]prompt.Suggest, CacheFunc) {
+	c := getPachClient()
 	defer c.Close()
```

Comment:
in other places, the `defer c.Close()` was removed - is this intentional?  It doesn't look like any code gracefully cleans up the client in those other paths.

Reply:
Good catch, I removed this and added cleanup after the terminal exits.